agent/proxycfg/mesh_gateway.go

```diff
@@ -258,6 +258,10 @@ func (s *handlerMeshGateway) handleUpdate(ctx context.Context, u UpdateEvent, sn
 				//                 Do those endpoints get cleaned up some other way?
 				delete(snap.MeshGateway.WatchedServices, sid)
 				cancelFn()
+
+				if len(snap.MeshGateway.ServiceGroups) > 0 {
```

Comment:
Is there a reason why this is wrapped with a length check?

Reply:
it was used for debugging. Removed. Thanks for reviewing.